lib/readiness.js

```diff
@@ -38,6 +40,29 @@ bedrock.events.on('bedrock.start', async () => {
   }
 });
 
+// the readiness checks defined in this module are bootstrapped during the
+// `bedrock.start` event which fires before the `bedrock.ready` event which is
+// responsible for triggering the `bedrock-server.readiness-check` event
+bedrock.events.on('bedrock-server.readiness-check', async () => {
+  // retry forever at the specified interval until the readiness check succeeds
+  await pWaitFor(async () => {
```

Comment:
This should have an [AbortController](https://developer.mozilla.org/en-US/docs/Web/API/AbortController), ([example declaration](https://github.com/digitalbazaar/bedrock-meter-usage-reporter/blob/initial/lib/meters.js#L21) and [example usage](https://github.com/digitalbazaar/bedrock-meter-usage-reporter/blob/initial/lib/meters.js#L385-L386)) that can be called on `bedrock.exit` ([example](https://github.com/digitalbazaar/bedrock-meter-usage-reporter/blob/initial/lib/meters.js#L63-L74)).

Otherwise, when bedrock receives a SIGTERM during this event, bedrock could wait forever.

Reply:
@dlongley One is able to terminate bedrock using ctrl-c SIGINT when inside this loop.  Does that not imply that SIGTERM will work properly as well?  I'm not sure how Bedrock/Nodejs would wait forever under these circumstances.

Furthermore, if I understand correctly, your solution precludes the use of `p-wait-for` and involves using the helper you've referenced, correct?  If that is the case, and if indeed we need to adopt this `AbortController` pattern more widely, I think we should find/create a module that has this feature.

Reply:
@mattcollier, you could still use `p-wait-for`, you'd just check the signal before rescheduling and throw an `AbortError` if it has been tripped. Some of the p-* libs already accept signals though (such as the delay one), so maybe another one would be better.

As for the SIGINT issue -- I guess what I should have said is that this does not terminate gracefully when SIGINT happens. What I would expect to happen is that `bedrock.exit` is emitted -- nothing handles it in this module and then `process.exit()` is called by `bedrock` which kills whatever is running here abruptly. I suppose that's acceptable, but it seems that it would be better to gracefully stop the loop. Since I guess it's "ok", this comment can be converted into an issue for someone to make it cleaner later.

Reply:
As for making another module, I'm not sure that would be all that helpful -- maybe there's something that can be reused, but it will mostly be whatever `AbortController` already offers ... it's that "other module", I think.

Reply:
Issue created, resolving.